src/main/java/com/hrznstudio/titanium/api/multiblock/MultiblockTemplate.java

```diff
@@ -0,0 +1,167 @@
+            return false;
+        }
+
+        Template template = getTemplate();
+        List<Mirror> mirrorStates;
+        if (canBeMirrored()) {
+            mirrorStates = ImmutableList.of(Mirror.NONE, Mirror.FRONT_BACK);
+        } else {
+            mirrorStates = ImmutableList.of(Mirror.NONE);
+        }
+
+        mirrorLoop:
+        for (Mirror mirror : mirrorStates) {
+            PlacementSettings placeSet = new PlacementSettings().setMirror(mirror).setRotation(rotation);
+            BlockPos origin = controllerPos.subtract(Template.transformedBlockPos(placeSet, triggerPos));
+            for (Template.BlockInfo info : template.blocks.get(0)) {
+                BlockPos realRelPos = Template.transformedBlockPos(placeSet, info.pos);
+                BlockPos here = origin.add(realRelPos);
+                BlockState expected = info.state.mirror(mirror).rotate(rotation);
+                BlockState inWorld = world.getBlockState(here);
+
+                if (!Matcher.matches(expected, inWorld, world, here, additionalPredicates).isAllowed()) {
+                    continue mirrorLoop;
+                }
+            }
+            formStructure(world, origin, rotation, mirror, direction);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void formStructure(World world, BlockPos originPos, Rotation rotation, Mirror mirror, Direction direction) {
+        BlockPos truePos = BlockPosUtil.withSettingsAndOffset(originPos, masterPos, mirror, rotation);
+        for (Template.BlockInfo blockInfo : template.blocks.get(0)) {
+            BlockPos actualPos = BlockPosUtil.withSettingsAndOffset(truePos, blockInfo.pos, mirror, rotation);
+        }
+    }
+
+    protected abstract void replaceStructureBlock(Template.BlockInfo info, World world, BlockPos originPos, boolean mirrored, Direction clickDirection, Vec3i offsetFromMaster);
+
+    @Override
+    public void breakStructure(World world, BlockPos originPos, Direction direction, boolean mirrored) {
+        Mirror mirror = mirrored ? Mirror.FRONT_BACK : Mirror.NONE;
+        Rotation rot = FacingUtil.getRotationBetweenFacings(Direction.NORTH, direction);
+        if (rot != null) {
+            for (Template.BlockInfo info : getTemplate().blocks.get(0)) {
+                BlockPos actualPos = BlockPosUtil.withSettingsAndOffset(originPos, info.pos, mirror, rot);
+                prepareBlockForDisassembly(world, actualPos);
+                world.setBlockState(actualPos, info.state.mirror(mirror).rotate(rot));
+            }
+        }
+    }
+
+    protected void prepareBlockForDisassembly(World world, BlockPos originPos) {}
+
+    @Override
+    public boolean canRenderFormedMultiblock() {
+        return false;
+    }
+
+    @Override
+    public void renderFormedMultiblock() {
+    }
+
+    public ResourceLocation getId() {
+        return id;
+    }
+
+    @Nullable
+    private Template getTemplate() {
+        if (template == null) {
```

Comment:
You'll probably want to add some sorta other check here, as if things fail, it'll call to load over and over.